### ISSUE TITLE
ValueError when initialising structured masked array having datetime64

### ISSUE THREAD

**Issue by gerritholl** [numpy/numpy#4583]
When initialising a structured masked array with `empty`, `zeros`, or `ones`, in the `numpy.ma` module, a `ValueError` is raised, as illustrated below:

```
In [19]: E = numpy.ma.zeros(1, [("time", "M8[s]")])                                                                                                                                                  
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-19-fa42e8196770> in <module>()
----> 1 E = numpy.ma.zeros(1, [("time", "M8[s]")])

/home/gerrit/venv/python-3.4/lib/python3.4/site-packages/numpy/ma/core.py in __call__(self, a, *args, **params)
   7221             _extras[p] = params.pop(p)
   7222         # Get the result
-> 7223         result = self._func.__call__(a, *args, **params).view(MaskedArray)
   7224         if "fill_value" in common_params:
   7225             result.fill_value = _extras.get("fill_value", None)

/home/gerrit/venv/python-3.4/lib/python3.4/site-packages/numpy/ma/core.py in __array_finalize__(self, obj)
   2798         if self.dtype.names:
   2799             if self._fill_value is None:
-> 2800                 self._fill_value = _check_fill_value(None, self.dtype)
   2801         return
   2802 

/home/gerrit/venv/python-3.4/lib/python3.4/site-packages/numpy/ma/core.py in _check_fill_value(fill_value, ndtype)
    400             descr = ndtype.descr
    401             fill_value = np.array(_recursive_set_default_fill_value(descr),
--> 402                                   dtype=ndtype,)
    403         else:
    404             fill_value = default_fill_value(ndtype)

ValueError: Error parsing datetime string "?" at position 0
```

**Comment by jjhelmus** [comment]
This issue was fixed in 4bfcf4801ceedd0563e3648154fc49985729c442 part of PR #5455 which also closed gh-4476, a similar issue to this one.  This issue can be closed.

**Comment by charris** [comment]
Thanks for the heads up.
